mirgecom/eos.py

```diff
@@ -129,6 +130,10 @@ def get_internal_energy(self, temperature, *, mass, species_mass_fractions):
         """Get the fluid internal energy from temperature and mass."""
         raise NotImplementedError()
 
+    def get_species_enthalpies(self, cv: ConservedVars):
+        """Get specific enthalpies for each mixture species."""
+        raise NotImplementedError()
+
```

Comment:
Oh, I was hoping it would return a NotImplemented error for non-mixture EOS. That way I have a specific exception to catch.  It is OK for the diffusive heat flux routine to be called with a non-mixture EOS, but as I have it this exception is the way that the routine knows to return an alternate structure.

Reply:
I dunno... I guess I'm not a huge fan of allowing partially-implemented interfaces. I think it's easier to reason about interfaces if a given class either fully implements the interface or doesn't implement it at all. One alternate way to do what you're currently doing here would be to create, say, a `MixtureEOS` base class that inherits from `GasEOS` and adds the mixture-specific stuff (and have `PyrometheusMixture` inherit from it, but `IdealSingleGas` not). Then you could check for `isinstance(eos, MixtureEOS)` in `diffusive_heat_flux` and return accordingly.

Reply:
I like your suggested approach, and it occurred to me to do it this way but I didn't think it would fly in review. I'll toy with it for a minute.

Reply:
Taking a pass at doing the MixtureEOS base class in #484

Reply:
(64cb2e2) :)